### PR TITLE
Server generation support

### PR DESCRIPTION
**Pull request by sfackler**
The server interface for a service consists of a trait implemented to
handle the actual logic of the service, and a resource object which
wraps a trait implementation and handles the HTTP serialization and
deserialization glue.

### CODE REVIEW

File: conjure-http/src/server.rs [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+pub trait Resource<B, R>: Sized
+where
+    B: RequestBody,
+    R: VisitResponse,
+{
+    /// The resource's name.
+    const NAME: &'static str;
+
+    /// Returns the resource's HTTP endpoints.
+    // ideally this would be a &'static [Endpoint] once const fns become more powerful
```
Comment (jonsyu1) [inline conjure-http/src/server.rs]: Is there an issue we can link here?

Comment (sfackler) [reply conjure-http/src/server.rs]: Unclear: https://github.com/rust-lang/rust/issues/57563#issuecomment-489681952

File: conjure-object/src/safe_long.rs [hunk 2]
```diff
@@ -132,3 +147,24 @@ impl fmt::Display for BoundsError {
...
+enum ParseErrorInner {
+    Parse(ParseIntError),
+    Bounds(BoundsError),
+}
+
+/// The error returned after failing to parse a string into a `SafeLong`.
+#[derive(Debug, Clone)]
+pub struct ParseError(ParseErrorInner);
+
+impl fmt::Display for ParseError {
```
Comment (jonsyu1) [inline conjure-object/src/safe_long.rs]: Do you have thoughts on [enum-display-derive](https://github.com/ihrwein/enum-display-derive)? I'm fine with this explicit implementation but I'm curious if there is an explicit reason not to use this crate.

Comment (sfackler) [reply conjure-object/src/safe_long.rs]: Doesn't seem worth it to pull in an extra dependency for one type.

**Pull request merged**